content/Software and Downloads/Upload/Error-avrdude-when-uploading.md

```diff
@@ -2,58 +2,87 @@
+* [avrdude: ser_open(): can't open device "portName": the system cannot find the file specified](#check-the-following)
+* [avrdude: ser_open(): can't open device "portName": the semaphore timeout period has expired.`](#check-the-following)
+-->
+
+> **Note:** The error text may be different depending on your system language.
 
 ---
 
-<h2 id="check-the-following">Check the following</h2>
+<a name="check-the-following"></a>
+
+## General troubleshooting checklist
+
+Most errors can be solved by following these steps. After each step, try uploading your sketch again.
+
+1. Make sure you have [selected the right board and port](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE) correct board under _Tools > Board_ and  _Tools > Port_.
 
-1. Make sure you have selected the correct board under `Tools > Board`.
+   * For instructions, see [Select board and port in Arduino IDE](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE)
 
-2. Make sure to select the correct port under `Tools > Port`.
+   * If your board is missing, see [If your board does not appear in the port menu](https://support.arduino.cc/hc/en-us/articles/4412955149586-If-your-board-does-not-appear-in-the-port-menu)
 
-3. Make sure the board is powered. Try disconnecting and reconnecting the board to the computer, and confirm that the power LED (usually labeled on the board "ON") is lit. On most boards, an orange LED will also blink when the board is plugged in.
+2. Close other instances of Arduino IDE, serial monitors or other software that my be blocking the port. If this doesn't work, try restarting your computer.
 
-4. Make sure that you are using a USB data cable (a charge-only cable will not work). Also, check the cable for physical damage.
+3. Reset your board by pressing the RESET button (a white or blue push button).
 
-5. Remove any connections to digital pins **0(RX)** and **1(TX)**. These pins are used for communication with your computer. Connecting anything to these pins can interfere with uploading. You can attach the components back to these pins once the program has been successfully uploaded.
+   ![The reset button on Arduino UNO.](img/uno-reset-button.png)
 
-6. Make sure that you have closed the serial monitor.
+4. Check port and cable connections:
 
-7. Press the reset button and see if the LED blinks. If the LED doesn't blink there is a high chance that the bootloader of your board is corrupted.
+   * Try disconnecting and reconnecting the board to the computer, and confirm that the power LED (usually labeled on the board "ON") is lit. On most boards, an orange LED will also blink when the board is plugged in. <!-- TODO: Simplify -->
 
-8. Make sure you have updated and installed all the necessary drivers for the board. If you already have the drivers, try reinstalling them.
+   * If possible, connect the board directly to a USB port. Avoid using a USB hub, or try a dif
```

Comment:
```suggestion
   * If possible, connect the board directly to a USB port. Avoid using a USB hub.
```
I don't know if there was supposed to be more information here.

Reply:
50973d56e1c6fb3498fd5c7952b66e3163155359